timemachine/fe/rbfe.py

```diff
@@ -125,34 +85,70 @@ def __init__(self, omm_system, conf, box):
         self.box = box
 
 
-def generate_samples(st, host_config, temperature, lambda_schedule, n_frames, seed):
+def sample(initial_state, protocol):
+    """
+    Generate a trajectory given an initial state and a simulation protocol
     """
-    Generate samples given a lambda_schedule. Let L be the number
-    of lambda windows T be the number of frames.
 
-    Parameters
-    ----------
-    st: SingleTopology
-        Topology container for the alchemical ligand
+    bound_impls = [p.bound_impl(np.float32) for p in initial_state.U_fns]
+    intg_impl = initial_state.integrator.impl()
+    baro_impl = initial_state.barostat.impl(bound_impls)
 
-    host_config: HostConfig
-        Configuration of the host
+    ctxt = custom_ops.Context(initial_state.x0, initial_state.v0, initial_state.box0, intg_impl, bound_impls, baro_impl)
 
-    temperature: float
-        Temperature of the system, in Kelvins
+    # burn-in
+    ctxt.multiple_steps_U(
+        lamb=initial_state.lamb,
+        n_steps=protocol.burn_in,
+        lambda_windows=[initial_state.lamb],
+        store_u_interval=0,
+        store_x_interval=0,
+    )
 
-    lambda_schedule: np.array(L,)
-        Sequence of lambda windows
+    # a crude, and probably not great, guess on the decorrelation time
+    n_steps = protocol.n_frames * protocol.steps_per_frame
+    all_nrgs, all_coords, all_boxes = ctxt.multiple_steps_U(
```

Comment:
Seems that we aren't using the energies, should we disable the lambda windows to avoid the additional cost?

Reply:
correct we're not using them right now, they were mostly for debugging purposes. These are computed once every 1000 steps right now, so it probably doesn't hurt to keep them.